packages/flutter_goldens/lib/flutter_goldens.dart

```diff
@@ -9,98 +9,319 @@ import 'package:file/file.dart';
+  @override
+  Future<void> update(Uri golden, Uint8List imageBytes) async {
+    final File goldenFile = getGoldenFile(golden);
+    await goldenFile.parent.create(recursive: true);
+    await goldenFile.writeAsBytes(imageBytes, flush: true);
+  }
+
+  /// Returns the golden file identified by the given [Uri].
+  @override
+  File getGoldenFile(Uri uri) {
+    return fs.directory(basedir).childFile(fs.file(uri).path);
+  }
+
+  /// Modifies the given [Uri] to incorporate an optional [version] number.
+  ///
+  /// Golden files for package:flutter use version numbers, and must be used to
+  /// reference the appropriate file.
+  @override
+  Uri getTestUri(Uri key, int version) {
+    return _addVersion(key, version);
+  }
+
+  Uri _addVersion(Uri key, int version) {
+    final String extension = path.extension(key.toString());
+    return version == null ? key : Uri.parse(
+      key
+        .toString()
+        .split(extension)
+        .join() + '.' + version.toString() + extension
+    );
+  }
+}
+
+/// Implementation of the abstract base class [FlutterGoldenFileComparator] for
+/// testing against with Skia Gold.
+///
+/// For testing across all platforms, the [SkiaGoldClient] is used to upload
+/// images for framework-related golden tests and process results. Currently
+/// these tests are designed to be run post-submit on Cirrus CI, informed by the
+/// environment.
+class FlutterSkiaGoldFileComparator implements FlutterGoldenFileComparator {
+  /// Creates a [FlutterSkiaGoldFileComparator] that will resolve golden file
+  /// URIs relative to the specified [basedir].
+  ///
+  /// The [fs] parameter exists for testing purposes only.
+  @visibleForTesting
+  FlutterSkiaGoldFileComparator(
+    this.basedir, {
+      this.fs = const LocalFileSystem(),
+      this.platform = const LocalPlatform(),
+    }) : assert(basedir != null);
+
+  /// The directory to which golden file URIs will be resolved in [compare] and
+  /// [update].
+  final Uri basedir;
+
+  /// The file system used to perform file access.
+  @visibleForTesting
+  final FileSystem fs;
+
+  /// A wrapper for the [dart:io.Platform] API.
+  ///
+  /// This is only for use in tests, where the system platform (the default) can
+  /// be replaced by mock platform instance.
+  @visibleForTesting
+  final Platform platform;
+
+  final SkiaGoldClient _skiaClient = SkiaGoldClient();
+
+  /// Creates a new [FlutterSkiaGoldFileComparator] that mirrors the relative
+  /// path resolution of the default [goldenFileComparator].
+  ///
+  /// The [goldens] and [defaultComparator] parameters are visible for testing
+  /// purposes only.
+  static Future<FlutterSkiaGoldFileComparator> fromDefaultComparator({
+    SkiaGoldClient goldens,
+    LocalFileComparator defaultComparator,
+  }) async {
+    defaultComparator ??= goldenFileComparator;
+    goldens ??= SkiaGoldClient();
+
+    // Calculate the appropriate basedir for the current test context.
+    final FileSystem fs = goldens.fs;
```

Comment:
Are these lines the same as for the other comparator? Consider implementing a protected method in the superclass that you can call from both sub-classes to do this work.